src/main.c

```diff
@@ -52,19 +52,19 @@ int main(int argc, char *argv[])
 		return 1;
 	}
 
-	// Define energies	
-	energies = malloc(sizeof(double)*n_energies);
-	if(energies == NULL){
-		printf("main: could not allocate memory for energies\n");
-		return 1;
-	}
-	for(i=0; i<n_energies; i++){
-		energies[i] = 1.+0.1*i;
-	}
+//	// Define energies	
+//	energies = malloc(sizeof(double)*n_energies);
+//	if(energies == NULL){
+//		printf("main: could not allocate memory for energies\n");
+//		return 1;
+//	}
+//	for(i=0; i<n_energies; i++){
+//		energies[i] = 1.+0.1*i;
+//	}
 
 	// Perform calculations	
 	printf("Starting calculations...\n");
-	efficiencies = polycap_source_get_transmission_efficiencies(source, nthreads, n_energies, energies, n_photons, NULL, &error);
+	efficiencies = polycap_source_get_transmission_efficiencies(source, nthreads, source->n_energies, source->energies, n_photons, NULL, &error);
```

Comment:
Ik denk dat je best `n_energies` and `energies` uit `polycap_source_get_transmission_efficiencies ` haalt als argumenten want ze zitten toch al in `source`.

Reply:
Idd, maar was niet zeker om dit te doen aangezien ik me half meende te herinneren dat je in het begin van onze samenwerking liever de energie etc afzonderlijk van source had, dus dacht dat dit mss een nuttige tussenoplossing was ;)

Maar zal het inderdaad aanpassen, houdt meer steek om enkel source mee te geven.

Reply:
Voor XMI-MSIM maakt het niet uit, aangezien we toch zullen werken met individuele `polycap_photon`'s en niet met een `polycap_source`.

Het lijkt me hoe dan ook logisch dat de beschrijving van het energie spectrum deel uitmaakt van de `source` 😄